tool/tsh/common/tsh_test.go

```diff
@@ -4898,18 +4898,16 @@ func TestSerializeMFADevices(t *testing.T) {
 	aTime := time.Date(1970, time.January, 1, 0, 0, 0, 0, time.UTC)
 	expected := `
 	[
-		{"metadata":{"Name":"my device"},"id":"id","addedAt":"1970-01-01T00:00:00Z","lastUsed":"1970-01-01T00:00:00Z"}
+		{"metadata":{"Name":"my device","Namespace":"default"},"id":"id","addedAt":"1970-01-01T00:00:00Z","lastUsed":"1970-01-01T00:00:00Z","kind":"mfa_device","totp":{"key":"key"},"version":"v1"}
```

Comment:
Why did we get changes here?

Are we suppose to serialize the TOTP key too?

Reply:
> Why did we get changes here?

Previously, `NewMFADevice` was not returning a valid MFA device (no `CheckAndSetDefaults`).

```go
// NewMFADevice creates a new MFADevice with the given name. Caller must set
// the Device field in the returned MFADevice.
func NewMFADevice(name, id string, addedAt time.Time) *MFADevice {
	return &MFADevice{
		Metadata: Metadata{
			Name: name,
		},
		Id:       id,
		AddedAt:  addedAt,
		LastUsed: addedAt,
	}
}
```

> Are we suppose to serialize the TOTP key too?

Yes, for entering it in storage on the backend. On the frontend the device would be returned without the key (`WithoutSensitiveData`). Double checked that this doesn't have any unexpected consequences.

Reply:
Thanks for double-checking.